### ISSUE TITLE
[setup] Automate pushing Storybook build up to GH-pages

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#4823]
To share work with UX and others periodically, let's push up our Storybook build for others to easily view.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-1542)
┆Issue Number: FXA-1542

**Comment by data-sync-user** [comment]
➤ Les Orchard commented:

Some context from when I last did this sort of thing for Test Pilot:

  * We used this script in CircleCI: https://github.com/mozilla/testpilot/blob/master/bin/deploy-storybook.sh ( https://github.com/mozilla/testpilot/blob/master/bin/deploy-storybook.sh )
  * The script does a few things:
    1. Run a static Storybook build and rename the directory based on the commit ID used in the current CI run.
    2. Adjust paths in the Storybook build - it expects to be at the root of the domain from which it's served up, but we published in sub-directories per commit hash so that we could compare between PRs
    3. Deploy to an AWS S3 bucket. (We can change this to a directory in the gh-pages branch)
    4. Generate a stub redirect page for pull request numbers to point at the directory named for the commit hash
    5. Post a comment via the github API to the pull request linking to the generated Storybook build

We currently generate and publish a Storybook build for payments-server along with other doco in this script: https://github.com/mozilla/fxa/blob/c1a0198946f9efdcefe3f381b2d7af3423e543f5/_scripts/gh-pages.sh#L22 ( https://github.com/mozilla/fxa/blob/c1a0198946f9efdcefe3f381b2d7af3423e543f5/_scripts/gh-pages.sh#L22 )

That script might be our best integration point. It does not do a build per commit for payments-server like we did on Test Pilot, though.

A procedural caveat will be that we can only publish these builds for PRs on the main Mozilla repo. That's because publishing to either S3 or gh-pages will require secrets that are not revealed to PRs from personal forks.

Another monorepo wrinkle might be that we put in a check for whether or not settings redesign files were modified in a PR. If not, don't run this Storybook script.

**Comment by data-sync-user** [comment]
➤ Les Orchard commented:

Here's an example github commit posted by the Storybook deploy script: https://github.com/mozilla/testpilot/pull/3638#issuecomment-400064647 ( https://github.com/mozilla/testpilot/pull/3638#issuecomment-400064647 )

The links are broken now, though, because we had a policy to delete old builds from S3 after 90 days. That might be less of a concern if it's in the gh-pages branch

**Comment by lmorchard** [comment]
Maybe as a note-to-self unless someone else picks this up: The folks behind Storybook have launched a service that essentially does what we want for this issue, plus some process tools for review & approval. That turns this into a build-vs-buy question, I think.

https://www.chromatic.com

**Comment by jodyheavener** [comment]
Possibly a naive observation, but I was looking at [storybook-deployer](https://github.com/storybookjs/storybook-deployer) and it makes it look fairly straightforward to deploy, with examples for doing it as a CI job as well. Is this an option? Or is it more involved somehow, possibly due to this being a monolithic repo?

**Comment by data-sync-user** [comment]
➤ Les Orchard commented:

Yeah, I've tried storybook-deployer before. It didn't seem to help with adjusting paths in the static build for being hosted at a sub-directory, which was our solution for publishing storybooks per-commit / per-PR at a single host name (i.e. an AWS S3 bucket configured for web hosting). I want to say it also clobbered over other storybooks already published so that only the new build ended up published.

That said, I think it's been a few years since I last looked at storybook-deployer, so a) it might be worth another look and b) what we did on Test Pilot might be worth doing differently
